integration-cli/docker_api_create_test.go

```diff
@@ -1,136 +0,0 @@
-		"Image": "busybox",
-		"Healthcheck": map[string]interface{}{
-			"Interval": -10 * time.Millisecond,
-			"Timeout":  time.Second,
-			"Retries":  int(1000),
-		},
-	}
-
-	res, body, err := request.Post("/containers/create?name="+name, request.JSONBody(config))
-	c.Assert(err, check.IsNil)
-	if versions.LessThan(testEnv.DaemonAPIVersion(), "1.32") {
-		c.Assert(res.StatusCode, check.Equals, http.StatusInternalServerError)
-	} else {
-		c.Assert(res.StatusCode, check.Equals, http.StatusBadRequest)
-	}
-
-	buf, err := request.ReadBody(body)
-	c.Assert(err, checker.IsNil)
-
-	expected := fmt.Sprintf("Interval in Healthcheck cannot be less than %s", container.MinimumDuration)
-	c.Assert(getErrorMessage(c, buf), checker.Contains, expected)
-
-	// test invalid Interval in Healthcheck: larger than 0s but less than 1ms
-	name = "test2"
-	config = map[string]interface{}{
-		"Image": "busybox",
-		"Healthcheck": map[string]interface{}{
-			"Interval": 500 * time.Microsecond,
-			"Timeout":  time.Second,
-			"Retries":  int(1000),
-		},
-	}
-	res, body, err = request.Post("/containers/create?name="+name, request.JSONBody(config))
-	c.Assert(err, check.IsNil)
-
-	buf, err = request.ReadBody(body)
-	c.Assert(err, checker.IsNil)
-
-	if versions.LessThan(testEnv.DaemonAPIVersion(), "1.32") {
-		c.Assert(res.StatusCode, check.Equals, http.StatusInternalServerError)
-	} else {
-		c.Assert(res.StatusCode, check.Equals, http.StatusBadRequest)
-	}
-	c.Assert(getErrorMessage(c, buf), checker.Contains, expected)
-
-	// test invalid Timeout in Healthcheck: less than 1ms
-	name = "test3"
-	config = map[string]interface{}{
-		"Image": "busybox",
-		"Healthcheck": map[string]interface{}{
-			"Interval": time.Second,
-			"Timeout":  -100 * time.Millisecond,
-			"Retries":  int(1000),
-		},
-	}
-	res, body, err = request.Post("/containers/create?name="+name, request.JSONBody(config))
-	c.Assert(err, check.IsNil)
-	if versions.LessThan(testEnv.DaemonAPIVersion(), "1.32") {
-		c.Assert(res.StatusCode, check.Equals, http.StatusInternalServerError)
-	} else {
-		c.Assert(res.StatusCode, check.Equals, http.StatusBadRequest)
-	}
-
-	buf, err = request.ReadBody(body)
-	c.Assert(err, checker.IsNil)
-
-	expected = fmt.Sprintf("Timeout in Healthcheck cannot be less than %s", container.MinimumDuration)
-	c.Assert(getErrorMessage(c, buf), checker.Contains, expected)
-
-	// test invalid Retries in Healthcheck: less than 0
-	name = "test4"
-	config = map[string]interface{}{
-		"Image": "busybox",
-		"Healthcheck": map[string]interface{}{
-			"Interval": time.Second,
-			"Timeout":  time.Second,
-			"Retries":  int(-10),
-		},
-	}
-	res, body, err = request.Post("/containers/create?name="+name, request.JSONBody(config))
-	c.Assert(err, check.IsNil)
-	if versions.LessThan(testEnv.DaemonAPIVersion(), "1.32") {
-		c.Assert(res.StatusCode, check.Equals, http.StatusInternalServerError)
-	} else {
-		c.Assert(res.StatusCode, check.Equals, http.StatusBadRequest)
```

Comment:
I see the new test no longer tests if the correct status code is returned; can we add that back to the test? We've had situations where the status code inadvertently changed, and we we didn't catch the regression

Reply:
Thank you for the review. Fixed.